src/flask_ml/flask_ml_server/utils.py

```diff
@@ -0,0 +1,359 @@
+
+    input_schema = task_schema.inputs
+    parameters_schema = task_schema.parameters
+
+    input_schema_input_key_to_input_type = {inputt.key: inputt.input_type for inputt in input_schema}
+    parameters_schema_key_to_parameter_type = {
+        parameter.key: parameter.value.parameter_type for parameter in parameters_schema
+    }
+
+    assert list(input_type_hints.keys()) == list(
+        input_schema_input_key_to_input_type.keys()
+    ), f"Input schema and Typed Dict for inputs must have the same keys. Input schema keys: {input_schema_input_key_to_input_type.keys()} | Typed Dict keys: {input_type_hints.keys()}"
+    assert list(parameters_type_hints.keys()) == list(
+        parameters_schema_key_to_parameter_type.keys()
+    ), f"Parameter schema and Typed Dict for parameters must have the same keys. Parameter schema keys: {parameters_schema_key_to_parameter_type.keys()} | Typed Dict keys: {parameters_type_hints.keys()}"
+
+    for key in input_schema_input_key_to_input_type:
+        input_type_hint = input_type_hints[key]
+        input_type = input_schema_input_key_to_input_type[key]
+        match input_type:
+            case InputType.FILE:
+                assert (
+                    input_type_hint is FileInput
+                ), f"For key {key}, the input type is InputType.FILE, but the TypeDict hint is {input_type_hint}. Change to FileInput."
+            case InputType.DIRECTORY:
+                assert (
+                    input_type_hint is DirectoryInput
+                ), f"For key {key}, the input type is InputType.DIRECTORY, but the TypeDict hint is {input_type_hint}. Change to DirectoryInput."
+            case InputType.TEXT:
+                assert (
+                    input_type_hint is TextInput
+                ), f"For key {key}, the input type is InputType.TEXT, but the TypeDict hint is {input_type_hint}. Change to TextInput."
+            case InputType.TEXTAREA:
+                assert (
+                    input_type_hint is TextInput
+                ), f"For key {key}, the input type is InputType.TEXTAREA, but the TypeDict hint is {input_type_hint}. Change to TextInput."
+            case InputType.BATCHFILE:
+                assert (
+                    input_type_hint is BatchFileInput
+                ), f"For key {key}, the input type is InputType.BATCHFILE, but the TypeDict hint is {input_type_hint}. Change to BatchFileInput."
+            case InputType.BATCHTEXT:
+                assert (
+                    input_type_hint is BatchTextInput
+                ), f"For key {key}, the input type is InputType.BATCHTEXT, but the TypeDict hint is {input_type_hint}. Change to BatchTextInput."
+            case InputType.BATCHDIRECTORY:
+                assert (
+                    input_type_hint is BatchDirectoryInput
+                ), f"For key {key}, the input type is InputType.BATCHDIRECTORY, but the TypeDict hint is {input_type_hint}. Change to BatchDirectoryInput."
```

Comment:
A dictionary can be used to remove these case statements. Code will be shorter. At least the main logic will be shorter and this will be much more readable.

```
input_type_validation = {
    InputType.FILE: FileInput,
    InputType.DIRECTORY: DirectoryInput,
    ...
}
assert (input_type_hint is input_type_validation[input_type], ...)
```